lib/content-services/src/lib/search/components/search-filter-chips/search-facet-chip-tabbed/search-facet-tabbed-content.component.spec.ts

```diff
@@ -189,15 +190,21 @@ describe('SearchFacetTabbedContentComponent', () => {
         spyOn(searchFacetService, 'updateSelectedBuckets').and.callThrough();
         component.submitValues();
         expect(component.submitValues).toHaveBeenCalled();
-        expect(queryBuilder.update).toHaveBeenCalled();
+        expect(queryBuilderUpdateSpy).toHaveBeenCalled();
         expect(component.updateDisplayValue).toHaveBeenCalled();
         expect(searchFacetService.updateSelectedBuckets).toHaveBeenCalled();
     });
 
     it('should update search query and display value on reset', () => {
         spyOn(component, 'updateDisplayValue').and.callThrough();
         component.reset();
-        expect(queryBuilder.update).toHaveBeenCalled();
+        expect(queryBuilderUpdateSpy).toHaveBeenCalled();
         expect(component.updateDisplayValue).toHaveBeenCalled();
     });
+
+    it('should not call queryBuilder.update on options change', () => {
+        expect(queryBuilderUpdateSpy.calls.count()).toBe(2);
+        component.onOptionsChange([{ value: 'test' }], 'field');
+        expect(queryBuilderUpdateSpy.calls.count()).toBe(2);
```

Comment:
I guess the logic behind the test is that since the number of calls to the function before onOptionsChange and after onOptionsChange is the same, then the function has not been called. However, I'm curious as to why the method is being called twice. I would have expected it to have been called 0 times (ideal) or once (maybe its needed during init).
Also, can you check and see if we can use toHaveBeenCalledBefore () here instead. That would ensure that if in the future the number of calls to queryBuilder.update change for whatever reason, then we won't have to update this test again. I guess the code might look something like this (untested) - 
```
it('should not call queryBuilder.update on options change', () => {
        spyOn(component, 'onOptionsChange').and.callThrough();
        component.onOptionsChange([{ value: 'test' }], 'field');
        expect(queryBuilderUpdateSpy).toHaveBeenCalledBefore(component.onOptionsChange);
});
```

Reply:
rewrote the test, thanks